### PR TITLE
String join() needs domain.first

### PR DESCRIPTION
**Pull request by psahabu**
@cassella caught this bug in ``String.join()``. linux64+flat testing pending.

### CODE REVIEW

**Comment by ben-albrecht** [comment]
Out of curiosity - does `domain.first` accomplish the same thing?

**Comment by psahabu** [comment]
I don't believe so? If the array had reverse indexing they might be different...?

**Comment by ben-albrecht** [comment]
Good point. Change looks good to me. 

I'll take this opportunity to document this as another potential use-case for the feature requested in #5368

**Comment by psahabu** [comment]
Actually, in the context of where this code is used, `first` works fine.

**Comment by bradcray** [comment]
While in a half-awake state this morning, I found myself thinking we should consider making low/high into procedures that take an optional 'param aligned=true' argument because I think the number of cases where we get this wrong is huge...

File: modules/internal/String.chpl [hunk 1]
```diff
@@ -835,7 +835,7 @@ module String {
         return '';
       } else if S.size == 1 {
         // TODO: ensures copy, clean up when no longer needed
-        var ret = S[S.domain.low];
```
Comment (ben-albrecht) [inline modules/internal/String.chpl]: This was a bug before (and still is), but this should not work for the tuple-case... I guess we don't have a test for `string.join(tuple)` anywhere.

**Pull request merged**